### ISSUE TITLE
Buggy ProductsInfoController

### ISSUE THREAD

**Issue by gerchicov-bp** [bizz84/SwiftyStoreKit#212]
### Platform

- [x] iOS

### In app purchase type

- [x] Auto-Renewable Subscription

### Environment

- [x] Sandbox
- [x] Production

### Version
0.9.0 (but the same code is not fixed in 0.9.2)

### Issue summary
1)request purchases
2)minimize app and change app store account to another one with different currency
3)return to app and request purchases again

### What did you expect to happen
Products return currency from the second account

### What happened instead
Products return currency from the first account. They are cached and the problem is with the whole `ProductsInfoController`:
`

guard products.count == productIds.count else {

            requestProducts(productIds, completion: completion)
            return
        }
`
1)`products.count == productIds.count` doesn't mean that all the products are the same - you should compare them by id
2)even if `products.count == productIds.count` then the currencies could be different as in my case
3)I wrote that the whole class is buggy because only `retrieveProductsInfo(...)` func in this class is public and other code is private so I can't even properly fix this code for my case.

So the fastest fix I see is to make everything inside `ProductsInfoController` public. The better fix is to allow force call `requestProducts` which is not called because of `products.count == productIds.count` statement

**Comment by numen31337** [comment]
The issue also appears if you change the price. There should be the way to refresh this data.

**Comment by bizz84** [comment]
The problem lies in the fact that SwiftyStoreKit caches products internally to that product information for any given product is only retrieved once.

Based on this report, I believe the best solution is to remove the caching altogether, and leave clients to implement it if desired.

This way, each time a new purchase is requested, SwiftyStoreKit will fetch the updated `SKProduct` that matches the chosen `productId`.

For convenience, in addition to these methods:

```swift
class func retrieveProductsInfo(_ productIds: Set<String>, ...)
class func purchaseProduct(_ productId: String, ...)
```

a new method could be made available:

```swift
class func purchaseProduct(_ product: SKProduct, ...)
```

This is so that if the client has already pre-fetched a product with `retrieveProductsInfo`, it can use it to make the purchase, without requesting the product information again.

I should have this implemented and working soon.

**Comment by gerchicov-bp** [comment]
@bizz84 wait. What about to store these methods and add something like the following:
```
class func retrieveProductsInfo(_ productIds: Set<String>, ..., forceUpdate: Bool)
{
  if forceUpdate {
    ...//clear cache
  }
  retrieveProductsInfo(...) //call to existing method
}
```

Or even a property which enables/disables caching.

**Comment by bizz84** [comment]
I think removing the cache is more in line with the **lightweight** spirit of SwiftyStoreKit, so I'm more inclined to remove state rather than adding state.

```swift
class func purchaseProduct(_ product: SKProduct, ...)
```

Clients can use this new method if they wish to prevent any un-necessary network calls, or they can use the old API and things will still work.

**Comment by bizz84** [comment]
Resolved in release [0.10.2](https://github.com/bizz84/SwiftyStoreKit/releases/tag/0.10.2).

**Comment by bb-git** [comment]
I liked the caching. It's sad you removed it for this weird use cases.
